users/views.py

```diff
@@ -59,12 +64,135 @@ def ActivationMailer(request):
+        except:
+            return HttpResponse("Please set up email host details!", status=206)
+    else :
+        return HttpResponse("Invalid request!", status=400)
+
+
+
+
+def HashPass(password):
+    password=password.encode()
+    return bcrypt.hashpw(password,bcrypt.gensalt())
+
+
+
+
+@csrf_exempt
+@api_view(['POST'])
+def SetPasswordAndActivate(request,token):
+    if request.method == "POST":
+        try:
+            pw=request.data['password']
+            user_data=User.objects.get(verification_code=token)
+            if user_data.activated==False:
+                user_data.activated=True
+                user_data.password=HashPass(pw).decode()
+                user_data.save()
+                response={
+                    'status':'success',
+                    'code':status.HTTP_200_OK,
+                    'message':'Password set succesfully and now you are registered',
+                }
+                return Response(response)
+            else:
+                response={
+                'code':'status.HTTP_401_UNAUTHORIZED',
+                'message':'Token already used'}
+                return Response(response,status=401)  
+        except:
+            response={
+                    'code':'status.HTTP_401_UNAUTHORIZED',
+                    'message':'Invalid token or invalid request'
+                }
+            return Response(response,status=status.HTTP_401_UNAUTHORIZED)
+    else:
+        return HttpResponse("Invalid Request",status=400)
+
+
+@csrf_exempt
+def ResetPasswordEmail(request):
+    if request.method == "POST":
+        try:
+            roll_no = JSONParser().parse(request)
+            user_data = User.objects.get(roll = roll_no['roll'])
+            sender = EMAIL_HOST_USER
+            recipient = user_data.email
+            name = user_data.name
+            user_code = user_data.generate_verification_code()
+            user_link = ACTIVATION_LINK[1].format(code = user_code)
+            subject = ACTIVATION_SUBJECT[1]
+            body = ACTIVATION_BODY[1].format(name=name, link=user_link)
             send_mail(subject, body, sender, [recipient], fail_silently=False)
-            return redirect(ACTIVATION_REDIRECT)
+            return redirect(ACTIVATION_REDIRECT[1])
         except:
             return HttpResponse("Please set up email host details!", status=206)
     else :
         return HttpResponse("Invalid request!", status=400)
+
+
+
+def pass_checker(old,password):
+    return bcrypt.checkpw(old.encode(),password)
+
+
+@csrf_exempt
+@api_view(['POST'])
+def ResetPassword(request,token):
+    if request.method == "POST":
+        try:
+            new1=request.data['new_password1']
+            new2=request.data['new_password2']
+            old=request.data['old_password']
+            user_data=User.objects.get(verification_code=token)
+            password=(user_data.password)
+            password=password.encode()
+
+            print(pass_checker(old,password))
```

Comment:
Please remove this `print` statement if it was used while debugging.

Reply:
i will do it